packages/amplify-migration-e2e/src/assertions.ts

```diff
@@ -0,0 +1,298 @@
+  const { userPoolGroups } = gen1Meta.auth;
+  expect(userPoolGroups.service).toEqual('Cognito-UserPool-Groups');
+  expect(userPoolGroups.providerPlugin).toEqual('awscloudformation');
+  expect(userPoolGroups.dependsOn.length).toBe(1);
+  expect(userPoolGroups.dependsOn[0].category).toBe('auth');
+  expect(userPoolGroups.dependsOn[0].attributes.length).toBe(4);
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('UserPoolId');
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('AppClientIDWeb');
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('AppClientID');
+  expect(userPoolGroups.dependsOn[0].attributes).toContain('IdentityPoolId');
+}
+
+export async function assertDefaultGen1Setup(projRoot: string) {
+  const gen1Meta = getProjectMeta(projRoot);
+  const gen1Region = gen1Meta.providers.awscloudformation.Region;
+  const { gen1UserPoolId } = await assertUserPool(gen1Meta, gen1Region);
+  const { gen1FunctionName } = await assertFunction(gen1Meta, gen1Region);
+  const { gen1BucketName } = await assertStorage(gen1Meta, gen1Region);
+  const { gen1GraphqlApiId } = await assertAPI(gen1Meta, gen1Region);
+  const { gen1IdentityPoolId } = await assertIdentityPool(gen1Meta, gen1Region);
+  const { gen1ClientIds } = await assertUserPoolClients(gen1Meta, gen1Region);
+
+  return { gen1UserPoolId, gen1ClientIds, gen1IdentityPoolId, gen1FunctionName, gen1BucketName, gen1GraphqlApiId, gen1Region };
+}
+
+export async function assertAuthWithMaxOptionsGen1Setup(projRoot: string) {
+  const gen1Meta = getProjectMeta(projRoot);
+  const gen1Region = gen1Meta.providers.awscloudformation.Region;
+  const { gen1UserPoolId } = await assertUserPool(gen1Meta, gen1Region);
+  const { gen1FunctionName } = await assertFunction(gen1Meta, gen1Region);
+  const { gen1IdentityPoolId } = await assertIdentityPool(gen1Meta, gen1Region);
+  const { gen1ClientIds } = await assertUserPoolClients(gen1Meta, gen1Region);
+  await assertUserPoolGroups(gen1Meta);
+
+  return { gen1UserPoolId, gen1ClientIds, gen1IdentityPoolId, gen1FunctionName, gen1Region };
+}
+
+export async function assertStorageWithMaxOptionsGen1Setup(projRoot: string) {
+  const gen1Meta = getProjectMeta(projRoot);
+  const gen1Region = gen1Meta.providers.awscloudformation.Region;
+  const { gen1BucketName } = await assertStorage(gen1Meta, gen1Region);
+  const { gen1UserPoolId } = await assertUserPool(gen1Meta, gen1Region);
+  const { gen1ClientIds } = await assertUserPoolClients(gen1Meta, gen1Region);
+  const { gen1IdentityPoolId } = await assertIdentityPool(gen1Meta, gen1Region);
+
+  return { gen1UserPoolId, gen1ClientIds, gen1BucketName, gen1IdentityPoolId, gen1Region };
+}
+
+export async function assertUserPoolResource(projRoot: string, gen1UserPoolId: string, gen1Region: string, projType?: string) {
+  const gen1Resource = await getResourceDetails('AWS::Cognito::UserPool', gen1UserPoolId, gen1Region);
+  if (projType) assert(gen1Resource.LambdaConfig.PostConfirmation);
```

Comment:
the `projType` arg is confusing, you dont need it. instead:
if (gen1resource.lambdaconfig.PostConfirmation) {
  assert(gen2Resource.LambdaConfig.PostConfirmation)
}

the null check for resource1 should be in `assertAuthWithMaxOptionsGen1Setup`

Reply:
Done [6e5a4bd](https://github.com/aws-amplify/amplify-cli/pull/13978/commits/6e5a4bd3857d7670e742c334b41e90c760be8cc8)